components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/internal/IdPManagementServiceComponent.java

```diff
@@ -316,21 +363,21 @@ protected void unsetIdentityProviderMgtListenerService(
     }
 
     private static Comparator<IdentityProviderMgtListener> idpMgtListenerComparator =
-            new Comparator<IdentityProviderMgtListener>(){
-
-        @Override
-        public int compare(IdentityProviderMgtListener identityProviderMgtListener1,
-                           IdentityProviderMgtListener identityProviderMgtListener2) {
-
-            if (identityProviderMgtListener1.getExecutionOrderId() > identityProviderMgtListener1.getExecutionOrderId()) {
-                return 1;
-            } else if (identityProviderMgtListener1.getExecutionOrderId() < identityProviderMgtListener1.getExecutionOrderId()) {
-                return -1;
-            } else {
-                return 0;
-            }
-        }
-    };
+            new Comparator<IdentityProviderMgtListener>() {
+
+                @Override
+                public int compare(IdentityProviderMgtListener identityProviderMgtListener1,
+                                   IdentityProviderMgtListener identityProviderMgtListener2) {
+
+                    if (identityProviderMgtListener1.getExecutionOrderId() > identityProviderMgtListener1.getExecutionOrderId()) {
```

Comment:
this should be
`identityProviderMgtListener1.getExecutionOrderId() > identityProviderMgtListener2.getExecutionOrderId()`
right?